### PR TITLE
ci: combine test and publish workflows

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

This combines `publish-next` with `test-main`, and `publish-geo` with `test-geo`. This will ensure all jobs show up in the checks UI:

image

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by reesscot — APPROVED** [review]
LGTM

**Review by ErikCH — APPROVED** [review]

File: .github/workflows/test-deploy-main.yml [hunk 1]
```diff
@@ -296,3 +296,30 @@ jobs:
           USERNAME: ${{ secrets.USERNAME }}
           NEW_PASSWORD: ${{ secrets.NEW_PASSWORD }}
           VALID_PASSWORD: ${{ secrets.VALID_PASSWORD }}
+
+  publish:
+    runs-on: ubuntu-latest
```
Comment (zchenwei) [inline .github/workflows/test-deploy-main.yml:301]: Should we only publish to next when all tests pass?

Comment (wlee221) [reply .github/workflows/test-deploy-main.yml:301]: Oh man, good catch! Let me make this more explicit.

Comment (wlee221) [reply .github/workflows/test-deploy-main.yml:302]: Done here [6782fe1](https://github.com/aws-amplify/amplify-ui/pull/1746/commits/6782fe17d2c768194dae31f68b82757963b4932e)

Comment (zchenwei) [reply .github/workflows/test-deploy-main.yml:302]: Awesome!

**Review by zchenwei — APPROVED** [review]
LGTM

**Pull request merged**